Bug Fix: Make call of literal_eval optional

hzeus opened the pull request:
<!--
# Thank you for contributing to LangChain-google!
-->

<!--
## Checklist for PR Creation

- [ ] PR Title: "[package]: [brief description]"

  - Where "package" is genai, vertexai, or community
  - Use "docs: ..." for purely docs changes, "templates: ..." for template changes, "infra: ..." for CI changes
  - Example: "community: add foobar LLM"

- [ ] PR Description and Relevant issues:

  - Description of the change
  - Relevant issues (if applicable)
  - Any dependencies required for this change

- [ ] Add Tests and Docs:

  - If adding a new integration:
    1. Include a test for the integration (preferably unit tests that do not rely on network access)
    2. Add an example notebook showing its use (place in the `docs/docs/integrations` directory)

- [ ] Lint and Test:
  - Run `make format`, `make lint`, and `make test` from the root of the package(s) you've modified
  - See contribution guidelines for more: https://github.com/langchain-ai/langchain-google/blob/main/README.md#contribute-code
-->

<!--
## Additional guidelines

- [ ] PR title and description are appropriate
- [ ] Necessary tests and documentation have been added
- [ ] Lint and tests pass successfully
- [ ] The following additional guidelines are adhered to:
  - Optional dependencies are imported within functions
  - No unnecessary dependencies added to pyproject.toml files (except those required for unit tests)
  - PR doesn't touch more than one package
  - Changes are backwards compatible
-->

## PR Description

Add a flag to make the call of `literal_eval` optional

## Relevant issues

Fixes #710 

## Type

🐛 Bug Fix


## Note(optional)

Hi 👋 

This is my attempt to make the usage of `literal_eval` optional. I added a flag to the ChatModel and set the default to `True` to keep it backwards compatible - however out of a security perspective I would rather suggest setting it to `False`.

In our case (we do not use `ChatPromptTemplate`), the usage of `literal_eval` is not necessary, but even leads to a bug when entering a value that is parseable as a Float ("3.14") into our chat (`TypeError: 'float' object is not iterable`)

lkuligin commented on libs/vertexai/tests/integration_tests/test_chat_models.py:
```diff
@@ -344,7 +344,9 @@ def test_parse_history_gemini_multimodal_FC():
-        history=history, imageBytesLoader=imageBytesLoader
+        history=history,
+        imageBytesLoader=imageBytesLoader,
+        perform_literal_eval_on_string_raw_content=True,
```
nits: the default is True anyways, do we really need it here?

maybe a unit test would be nice that tests the desired behaviour

hzeus commented:
Thanks for your review!

I added two unit tests for `_parse_chat_history_gemini` and also adjusted the snapshot which should fix the test runs.

> nits: the default is True anyways, do we really need it here?

I set the default to `True` to the initialization of the Model, but not for the direct call of the function.

hzeus commented:
I just noticed that the previous commit failed the linter. Hopefully fixed that.
